boa_gc/src/internals/gc_box.rs

```diff
@@ -5,80 +5,54 @@ use std::{
     ptr::{self, NonNull},
 };
 
-// Age and Weak Flags
-const MARK_MASK: usize = 1 << (usize::BITS - 1);
-const ROOTS_MASK: usize = !MARK_MASK;
-const ROOTS_MAX: usize = ROOTS_MASK;
-
 /// The `GcBoxheader` contains the `GcBox`'s current state for the `Collector`'s
 /// Mark/Sweep as well as a pointer to the next node in the heap.
 ///
 /// These flags include:
-///  - Root Count
 ///  - Mark Flag Bit
 ///
 /// The next node is set by the `Allocator` during initialization and by the
 /// `Collector` during the sweep phase.
 pub(crate) struct GcBoxHeader {
-    roots: Cell<usize>,
+    marked: Cell<bool>,
```

Comment:
Any thoughts on using a bit in the `ref_count` here over a full bool?

Reply:
I was hesitant to keep the mark bit in `ref_count` since there was 1~2% overhead, but it seems like putting the mark bit into `non_root_count` does not have that much of overhead. Updated to include the mark bit in `non_root_count`.